### DISCUSSION TITLE
how to remove a dinit service?

### DISCUSSION THREAD

**Discussion by aanderse** [davmac314/dinit#369] [General]
hello :wave:

as a concrete example:

```
-bash-5.2# dinitctl list
[[+]     ] boot
[{+}     ] atd (pid: 75)
...

-bash-5.2# rm /etc/dinit.d/atd

-bash-5.2# dinitctl list
[[+]     ] boot
[     {-}] atd
```

how do i properly reload `dinit` so it understands `atd` doesn't exist anymore?

for additional context - i want to entirely replace the `/etc/dinit.d` directory during runtime (system operation) where some of the files will remain the same, some will be removed, some will be added, and some will be modified... anything goes (to a reasonable extent)
assume i have a list of all added services, removed services, and modified services so i can run `dinitctl <action> <service>` for each and every change

thank you kindly :bowing_man:

**Comment by mobin-2008** [comment]
Hi.

`dinitctl unload atd`.

From [`dinitctl(8)`](https://davmac.org/projects/dinit/man-pages-html/dinitctl.8.html#unload) man-page:
> Completely unload a service. This can only be done if the service is stopped and has no loaded dependents (i.e. dependents must be unloaded before their dependencies).

Also check out other commands such as `reload` in that man-page.

Regards

**Reply by aanderse** [reply]
thank you very much, i missed that man page :sweat_smile:
